runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -243,13 +243,7 @@ public void gc(long trimMark) {
             // Check if direct access is possible (unlikely).
             if (directAccessCheckFunction.apply(this)) {
                 log.trace("Access [{}] Direct (writelock) access at {}", this, getVersionUnsafe());
-                R ret = accessFunction.apply(object.getContext(ICorfuExecutionContext.DEFAULT));
-
-                long versionForCorrectness = getVersionUnsafe();
-                if (lock.validate(ts)) {
-                    correctnessLogger.trace("Version, {}", versionForCorrectness);
-                    return ret;
-                }
+                return accessFunction.apply(object.getContext(ICorfuExecutionContext.DEFAULT));
```

Comment:
I don't think you need to check if direct access is possible since it is called right after the write lock is acquired. 
Remove it?

Its possible that from the time the optimistic read happened to the time the lock was acquired the version can change back to the original access version, but this if-call is not necessary because if the VLO is at the right version then `updateFunction.accept(this);` will just be a no-op

Reply:
Discussed offline, this can be an optimization in non-transactional read cases.